core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -734,9 +734,20 @@ public BytesReference ext() {
         return ext;
     }
 
+    /**
+     * Create a new SearchSourceBuilder with attributes set by an xContent.
+     */
     public SearchSourceBuilder fromXContent(XContentParser parser, QueryParseContext context, AggregatorParsers aggParsers)
             throws IOException {
         SearchSourceBuilder builder = new SearchSourceBuilder();
+        builder.parseXContent(parser, context, aggParsers);
```

Comment:
Hmmm, ok that makes more sense. I didn't see that before. I'm fine with doing it this way then and +1 to adding a comment to `parseXContent()` so if future features need to know where to do the same thing they can. If you could add a small comment to the merge commit too that would be great